packages/lib/src/__tests__/format.test.ts

```diff
@@ -0,0 +1,224 @@
+	const buildOutputFormatterSpy = jest.spyOn(outputBuilder, 'buildOutputFormatter')
+	const writeOutputSpy = jest.spyOn(output, 'writeOutput')
+
+	beforeEach(() => {
+		outputFormatter.mockReturnValue('output')
+		buildOutputFormatterSpy.mockReturnValue(outputFormatter)
+		writeOutputSpy.mockResolvedValue()
+	})
+
+	afterEach(() => {
+		jest.clearAllMocks()
+	})
+
+	describe('formatAndWriteItem', () => {
+		let itemTableFormatterSpy: jest.SpyInstance<output.OutputFormatter<unknown>, [tableGenerator: TableGenerator, fieldDefinitions: TableFieldDefinition<unknown>[]]>
+
+		beforeEach(() => {
+			itemTableFormatterSpy = jest.spyOn(output, 'itemTableFormatter')
+		})
+
+		it('uses tableFieldDefinitions when specified', async () => {
+			const command = {
+				...baseCommand,
+				tableFieldDefinitions: [],
+			}
+
+			const commonFormatter = jest.fn()
+			itemTableFormatterSpy.mockReturnValue(commonFormatter)
+
+			await formatAndWriteItem(command, item, IOFormat.COMMON)
+
+			expect(itemTableFormatterSpy).toHaveBeenCalledTimes(1)
+			expect(itemTableFormatterSpy).toHaveBeenCalledWith(command.tableGenerator, command.tableFieldDefinitions)
+			expect(buildOutputFormatterSpy).toHaveBeenCalledTimes(1)
+			expect(buildOutputFormatterSpy).toHaveBeenCalledWith(command, IOFormat.COMMON, commonFormatter)
+			expect(outputFormatter).toHaveBeenCalledTimes(1)
+			expect(outputFormatter).toHaveBeenCalledWith(item)
+			expect(writeOutputSpy).toHaveBeenCalledTimes(1)
+			expect(writeOutputSpy).toHaveBeenCalledWith('output', 'output.yaml')
+		})
+
+		it('uses buildTableOutput when specified', async () => {
+			const command = {
+				...baseCommand,
+				buildTableOutput: jest.fn(),
+			}
+
+			await formatAndWriteItem<SimpleType>(command, item, IOFormat.JSON)
+
+			expect(itemTableFormatterSpy).toHaveBeenCalledTimes(0)
+			expect(buildOutputFormatterSpy).toHaveBeenCalledTimes(1)
+			expect(buildOutputFormatterSpy).toHaveBeenCalledWith(command, IOFormat.JSON, expect.anything())
+			expect(outputFormatter).toHaveBeenCalledTimes(1)
+			expect(outputFormatter).toHaveBeenCalledWith(item)
+			expect(writeOutputSpy).toHaveBeenCalledTimes(1)
+			expect(writeOutputSpy).toHaveBeenCalledWith('output', 'output.yaml')
+
+			// Call the OutputFormatter that was build to ensure it uses `buildTableOutput`
+			const commonFormatter: output.OutputFormatter<SimpleType> = buildOutputFormatterSpy.mock.calls[0][2] as never
+			command.buildTableOutput.mockReturnValue('common output')
+			expect(commonFormatter(item)).toBe('common output')
+			expect(command.buildTableOutput).toHaveBeenCalledTimes(1)
+		})
+	})
+
+	describe('formatAndWriteList', () => {
+		let listTableFormatterSpy: jest.SpyInstance<output.OutputFormatter<unknown[]>, [tableGenerator: TableGenerator, fieldDefinitions: TableFieldDefinition<unknown>[], includeIndex?: boolean | undefined]>
+
+		beforeEach(() => {
+			listTableFormatterSpy = jest.spyOn(output, 'listTableFormatter')
```

Comment:
This can be moved out of beforeEach to spy initialization as well. I see `listTableFormatterSpy` gets its implementation mocked in select tests, so we would want to add a call to [restore](https://jestjs.io/docs/en/mock-function-api#mockfnmockrestore) it between tests (or call it at the end of the tests that set it).

Reply:
Fixed